core/src/test/java/kafka/admin/AdminFenceProducersIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+
+@ClusterTestDefaults(serverProperties = {
+        @ClusterConfigProperty(key = ServerLogConfigs.AUTO_CREATE_TOPICS_ENABLE_CONFIG, value = "false"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_PARTITIONS_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_REPLICATION_FACTOR_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionLogConfig.TRANSACTIONS_TOPIC_MIN_ISR_CONFIG, value = "1"),
+        @ClusterConfigProperty(key = TransactionStateManagerConfig.TRANSACTIONS_ABORT_TIMED_OUT_TRANSACTION_CLEANUP_INTERVAL_MS_CONFIG, value = "2000")
+})
+@ExtendWith(ClusterTestExtensions.class)
+public class AdminFenceProducersIntegrationTest {
+    private static final String TOPIC_NAME = "mytopic";
+    private static final String TXN_ID = "mytxnid";
+    private static final String INCORRECT_BROKER_PORT = "225";
+    private static final ProducerRecord<byte[], byte[]> RECORD = new ProducerRecord<>(TOPIC_NAME, null, new byte[1]);
+    private final ClusterInstance clusterInstance;
+    private Admin adminClient;
+    private KafkaProducer<byte[], byte[]> producer;
+
+    AdminFenceProducersIntegrationTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+    }
+
+    private KafkaProducer<byte[], byte[]> createProducer() {
+        Properties config = new Properties();
+        config.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers());
+        config.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, TXN_ID);
+        config.put(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, "2000");
+        config.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+        config.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+
+        return new KafkaProducer<>(config);
+    }
+
+    @ClusterTest
+    void testFenceAfterProducerCommit() throws Exception {
+        clusterInstance.createTopic(TOPIC_NAME, 1, (short) 1);
+
+        producer = createProducer();
+        producer.initTransactions();
+        producer.beginTransaction();
+        producer.send(RECORD).get();
+        producer.commitTransaction();
+
+        adminClient = clusterInstance.createAdminClient();
+        adminClient.fenceProducers(Collections.singletonList(TXN_ID)).all().get();
+
+        producer.beginTransaction();
+        try {
+            producer.send(RECORD).get();
+            fail("expected ProducerFencedException");
+        } catch (ProducerFencedException e) {
+            // expected
+        } catch (ExecutionException e) {
+            assertInstanceOf(ProducerFencedException.class, e.getCause());
+        }
+
+        assertThrows(ProducerFencedException.class, producer::commitTransaction);
+
+        adminClient.close();
+        producer.close();
+    }
+
+    @ClusterTest
+    @Timeout(30)
```

Comment:
The `@ClusterTest` has timeout with 60s, thus I think we didn't override timeout in this test, WDYT?